Flow everywhere

chrisnojima opened the pull request:
- [x] Adds `@flow` everywhere
- [x] Fixes 1 bug!
- [x] Cleans up package.json and lets our child scripts be like the rest of our code (flow/es6)
- [x] Icons are sorted now (stop dumb diffs)

chrisnojima commented on .eslintrc:
```diff
@@ -29,7 +29,7 @@
         "comma-dangle": [2, "always-multiline"],
         "filenames/match-regex": [2, "^[0-9a-z-.]+(\\.desktop|\\.native|\\.ios|\\.android)?$"],
-        "flowtype/require-valid-file-annotation": [2, "never"], // TODO always once we get 100% clean
+        "flowtype/require-valid-file-annotation": [2, "always"],
```
aww yisssss

chrisnojima commented on desktop/app/store-helper.js:
```diff
@@ -46,7 +47,7 @@ export default function (mainWindow) {
     // Reverse so the indexes don't shift
     dead.reverse().forEach(idx => {
-      subscribeStoreSubscribers.splice(dead, 1)
+      subscribeStoreSubscribers.splice(idx, 1)
```
this was a real bug! dead tracker windows likely weren't cleaning up correctly

chrisnojima commented on desktop/npm-helper.js:
```diff
@@ -38,11 +39,13 @@ function pad (s, num) {
   return s
 }
 
+const nodeCmd = 'babel-node --presets es2015,stage-2 --plugins transform-flow-strip-types'
```
have our child scripts be able to run w/ es2015 and flow

chrisnojima commented on desktop/npm-helper.js:
```diff
@@ -352,7 +357,7 @@ const iconMeta_ = {
   // eslint really doesn't understand embedded backticks
 /* eslint-disable */
-Object.keys(icons).map(name => {
+Object.keys(icons).sort().map(name => {
```
sort our icons to reduce the amount of churn in this script

chrisnojima commented on desktop/npm-vendor.js:
```diff
@@ -89,9 +96,9 @@ function updateVendored () {
   vendorURL = vendorURL.replace(/^git@github.com:/, 'https://github.com/')
   var vendorCommit = spawn('git', ['rev-parse', 'HEAD'], {cwd: VENDOR_DIR, stdio: 'pipe'}).stdout.trim()
-  var packageInfo = JSON.parse(fs.readFileSync('./package.json'))
+  var packageInfo = JSON.parse(fs.readFileSync('./package.json', {encoding: 'utf8'}))
```
dunno how this worked w/o the encoding. it should send back a buffer

chrisnojima commented on desktop/package.json:
```diff
@@ -4,31 +4,33 @@
-    "postinstall": "babel-node --presets es2015,stage-2 npm-helper.js postinstall",
-    "render-screenshots": "babel-node --presets es2015,stage-2 npm-helper.js render-screenshots",
-    "updated-fonts": "babel-node --presets es2015,stage-2 npm-helper.js updated-fonts",
+    "_node": "babel-node --presets es2015,stage-2 --plugins transform-flow-strip-types ",
```
some helpers to reduce all the copy paste

chrisnojima commented on shared/nav.android.js:
```diff
@@ -110,7 +113,7 @@ class Nav extends Component {
     const drawerContent = (
       <Box style={{flex: 1, backgroundColor: '#fff'}}>
-        <Text style={{margin: 10, fontSize: 15, textAlign: 'left'}}>I'm in the Drawer!</Text>
+        <Text type='Header' style={{margin: 10, fontSize: 15, textAlign: 'left'}}>I'm in the Drawer!</Text>
```
no type= before???

chrisnojima commented on shared/settings/about.js:
```diff
@@ -1,18 +0,0 @@
-import React, {Component} from 'react'
```
just killing all this old non-used settings junk 👊

chromakode replied:
:scissors: :heart:

keybase-ci-visdiff commented:
The commits 6371a058a8f2...e33c6309995b introduce visual changes on linux. <details><summary>:mag_right: 2 changed</summary>

 * Changed: **Icon-Icon_Image** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475588004672-6371a058a8f2-e33c6309995b-linux/Icon-Icon_Image-6371a058a8f2.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475588004672-6371a058a8f2-e33c6309995b-linux/Icon-Icon_Image-e33c6309995b.png)  
   ![Icon-Icon_Image comparison](https://keybase-app-visdiff.s3.amazonaws.com/1475588004672-6371a058a8f2-e33c6309995b-linux/Icon-Icon_Image.png)
 * Changed: **Icon-Icon_Icon_Font** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475588004672-6371a058a8f2-e33c6309995b-linux/Icon-Icon_Icon_Font-6371a058a8f2.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475588004672-6371a058a8f2-e33c6309995b-linux/Icon-Icon_Icon_Font-e33c6309995b.png)  
   ![Icon-Icon_Icon_Font comparison](https://keybase-app-visdiff.s3.amazonaws.com/1475588004672-6371a058a8f2-e33c6309995b-linux/Icon-Icon_Icon_Font.png)
</summary>

keybase-ci-visdiff commented:
The commits 6371a058a8f2...e33c6309995b introduce visual changes on linux. <details><summary>:mag_right: 2 changed</summary>

 * Changed: **Icon-Icon_Image** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475591298049-6371a058a8f2-e33c6309995b-linux/Icon-Icon_Image-6371a058a8f2.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475591298049-6371a058a8f2-e33c6309995b-linux/Icon-Icon_Image-e33c6309995b.png)  
   ![Icon-Icon_Image comparison](https://keybase-app-visdiff.s3.amazonaws.com/1475591298049-6371a058a8f2-e33c6309995b-linux/Icon-Icon_Image.png)
 * Changed: **Icon-Icon_Icon_Font** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475591298049-6371a058a8f2-e33c6309995b-linux/Icon-Icon_Icon_Font-6371a058a8f2.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475591298049-6371a058a8f2-e33c6309995b-linux/Icon-Icon_Icon_Font-e33c6309995b.png)  
   ![Icon-Icon_Icon_Font comparison](https://keybase-app-visdiff.s3.amazonaws.com/1475591298049-6371a058a8f2-e33c6309995b-linux/Icon-Icon_Icon_Font.png)
</summary>

chrisnojima commented:
@keybase/react-hackers please take a look

keybase-ci-visdiff commented:
The commits f666fb7a535b...afe17cc9a22f introduce visual changes on linux. <details><summary>:mag_right: 2 changed</summary>

 * Changed: **Icon-Icon_Image** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475618168565-f666fb7a535b-afe17cc9a22f-linux/Icon-Icon_Image-f666fb7a535b.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475618168565-f666fb7a535b-afe17cc9a22f-linux/Icon-Icon_Image-afe17cc9a22f.png)  
   ![Icon-Icon_Image comparison](https://keybase-app-visdiff.s3.amazonaws.com/1475618168565-f666fb7a535b-afe17cc9a22f-linux/Icon-Icon_Image.png)
 * Changed: **Icon-Icon_Icon_Font** [(before)](https://keybase-app-visdiff.s3.amazonaws.com/1475618168565-f666fb7a535b-afe17cc9a22f-linux/Icon-Icon_Icon_Font-f666fb7a535b.png) [(after)](https://keybase-app-visdiff.s3.amazonaws.com/1475618168565-f666fb7a535b-afe17cc9a22f-linux/Icon-Icon_Icon_Font-afe17cc9a22f.png)  
   ![Icon-Icon_Icon_Font comparison](https://keybase-app-visdiff.s3.amazonaws.com/1475618168565-f666fb7a535b-afe17cc9a22f-linux/Icon-Icon_Icon_Font.png)
</summary>

chromakode commented on desktop/npm-vendor.js:
```diff
@@ -1,11 +1,16 @@
 
-var VENDOR_DIR = process.env.KEYBASE_JS_VENDOR_DIR
+var VENDOR_DIR = process.env.KEYBASE_JS_VENDOR_DIR || ''
+if (!VENDOR_DIR) {
```
This duplicates a check near the bottom of the script. Should it be moved up?

chromakode commented on desktop/npm-vendor.js:
```diff
@@ -89,9 +96,9 @@ function updateVendored () {
+  var packageInfo = JSON.parse(fs.readFileSync('./package.json', {encoding: 'utf8'}))
   packageInfo.keybaseVendoredDependencies = `${vendorURL}#${vendorCommit}`
-  fs.writeFileSync('./package.json', JSON.stringify(packageInfo, 2, 2))
+  fs.writeFileSync('./package.json', JSON.stringify(packageInfo, null, 2))
```
Oops! Thanks `flow`.

chromakode commented on desktop/package.json:
```diff
@@ -4,32 +4,35 @@
-    "shrinkwrap-audit": "babel-node --presets es2015,stage-2 npm-shrinkwrap-audit",
-    "help": "babel-node --presets es2015,stage-2 npm-helper.js help",
-    "watch-test-file": "KEYBASE_RUN_MODE=devel babel-node --presets es2015,stage-2 npm-helper.js watch-test-file",
+    "_helper": "npm run _node -- npm-helper.js",
```
So much nicer!

chromakode commented on shared/profile/index.desktop.js:
```diff
@@ -1,15 +1,16 @@
 /* @flow */
```
`// @flow`?

chrisnojima replied:
yeah, i went ahead and changed that everywhere

chrisnojima commented:
@chromakode please take a look at the new commits. i added a couple of new rules so we have comma dangling like our regular js code, removed semicolons also. changed all /* style to // and addressed your PR feedback (also a master merge)
so close!

chrisnojima commented:
also ticketed an option to force this /* vs // style. https://github.com/gajus/eslint-plugin-flowtype/issues/126
